v0/destinations/hs/transform.js

```diff
@@ -212,6 +209,79 @@ function process(event) {
   return processSingleMessage(event.message, event.destination);
 }
 
+const batch = destEvents => {
+  const batchedResponse = [];
+
+  let i = 0;
+  const n = destEvents.length;
+  const arrayChunksIdentify = [];
+  let chunks = [];
+  while (i < n) {
+    if (destEvents[i].message.method === "GET") {
+      const { message, metadata, destination } = destEvents[i];
+      const endpoint = get(message, "endpoint");
+
+      const response = defaultBatchRequestConfig();
+      response.batchedRequest.headers = message.headers;
+      response.batchedRequest.endpoint = endpoint;
+      response.batchedRequest.body = message.body;
+      response.batchedRequest.params = message.params;
+      response.batchedRequest.method = defaultGetRequestConfig.requestMethod;
+      response.metadata = [metadata];
+      response.destination = destination;
+
+      batchedResponse.push(response);
+    } else {
+      chunks.push(destEvents[i]);
+    }
+    i += 1;
+    if (chunks.length && (chunks.length === MAX_BATCH_SIZE || i === n)) {
+      arrayChunksIdentify.push(chunks);
+      chunks = [];
+    }
+  }
+
+  arrayChunksIdentify.forEach(chunk => {
+    const identifyResponseBodyJson = [];
+    const metadata = [];
+
+    // extracting destination, apiKey value
+    // from the first event in a batch
+    const { destination } = chunk[0];
+    const { apiKey } = destination.Config;
+    const params = { hapikey: apiKey };
+
+    const endpoint = "https://api.hubapi.com/contacts/v1/contact/batch/";
+    let batchEventResponse = defaultBatchRequestConfig();
+
+    chunk.forEach(ev => {
+      const email = ev.message.body.JSON.properties[0].value;
```

Comment:
are we sure that email will always be the first object in the array?

Reply:
Yes